shared/lib/site-settings/exporter/reducers.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import Immutable from 'immutable';
+import debugModule from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TOGGLE_EXPORTER_ADVANCED_SETTINGS,
+	TOGGLE_EXPORTER_SECTION
+} from '../action-types';
+
+const debug = debugModule( 'calypso:exporter' );
+
+const initialUIState = Immutable.fromJS( {
+	advancedSettings: {
+		isVisible: false,
+		posts: {
+			isEnabled: true
+		},
+		pages: {
+			isEnabled: true
+		},
+		feedback: {
+			isEnabled: true
+		}
+	}
+} );
+
+/**
+ * Reducer for managing the exporter UI
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function ui( state = initialUIState, action ) {
+	switch ( action.type ) {
+		case TOGGLE_EXPORTER_SECTION:
+			debug( 'toggle section', action.section );
+			return state.updateIn( [ 'advancedSettings', action.section, 'isEnabled' ], ( x ) => ( !x ) );
+		case TOGGLE_EXPORTER_ADVANCED_SETTINGS:
+			debug( 'toggle advanced settings' );
+			return state.updateIn( [ 'advancedSettings', 'isVisible' ], ( x ) => ( !x ) );
+	}
+
+	return state;
+}
+
+export default combineReducers( {
+	ui
+} );
```

Comment:
There's a kind of hidden feature of calling `combineReducers` that isn't really obvious; It automatically creates a branch on the state tree for that reducer to manage. So you end up with a tree of:

    {
      siteSettings: {
        exporter: {
          ui: {
            advancedSettings: { ... }
          }
       }
    }

By handing back the `ui` reducer itself instead of wrapping it in the `combineReducers` call, the state tree would be:

    {
      siteSettings: {
        exporter: {
          advancedSettings: { ... }
       }
    }

 Of course since `ui` is currently the only reducer it would make sense to cut out that branch entirely, but in an upcoming PR I'll be adding another branch to manage the non-UI state/the data retrieved from the API, so I've left it in there for now.